pkg/querier/queryrange/views.go

```diff
@@ -0,0 +1,332 @@
+package queryrange
+
+import (
+	"fmt"
+	"io"
+	"sort"
+
+	"github.com/cespare/xxhash/v2"
+	jsoniter "github.com/json-iterator/go"
+	"github.com/richardartoul/molecule"
+	"github.com/richardartoul/molecule/src/codec"
+
+	"github.com/grafana/loki/pkg/logproto"
+	"github.com/grafana/loki/pkg/querier/queryrange/queryrangebase"
+)
+
+// GetLokiSeriesResponseView returns a view on the series response of a
+// QueryResponse. Returns and error if the message was empty. Note: the method
+// does not verify that the reply is a propberly encoded QueryResponse protobuf.
+func GetLokiSeriesResponseView(data []byte) (view *LokiSeriesResponseView, err error) {
+	b := codec.NewBuffer(data)
+	err = molecule.MessageEach(b, func(fieldNum int32, value molecule.Value) (bool, error) {
+		if fieldNum == 1 {
```

Comment:
Could we get the field index somehow to make it more clear which field we're referring to?

Reply:
You mean from the descriptor? Or would it be enough to declare a constant.

Reply:
Done